src/app/src/test/kotlin/org/vaccineimpact/orderlyweb/tests/unit_tests/controllers/web/ReportControllerTests/GetLatestReportVersionTests.kt

```diff
@@ -0,0 +1,434 @@
+package org.vaccineimpact.orderlyweb.tests.unit_tests.controllers.web.ReportControllerTests
```

Comment:
These tests are slightly amended versions of the `GetByNameAndVersionTests`.  Since the logic for the two controllers is near identical, except for the version line, I was not sure whether all the tests needed to be duplicated or if I should just add tests that cover the differences between the controllers.

Reply:
I think you've answered your own question - the code for the two methods is so similar that it would (in my opinion) make sense for it to be extracted into a new method that takes an optional version identifier and is called by both existing methods.

If you do that then you can rely on the underlying logic being mostly verified by the tests for the existing endpoint, and just add new test(s) to ensure that the `latest` endpoint does actually build a view model for the latest version. You can of course add a comment in the code to explain that.

Either way, I don't think it's sensible to essentially duplicate 400 lines of code as that just means more maintenance.